model_analyzer/device/gpu_device_factory.py

```diff
@@ -223,19 +228,47 @@ def get_cuda_visible_gpus(self):
             UUIDs of the DCGM supported devices visible to CUDA
         """
 
-        cuda_visible_gpus = []
-        if numba.cuda.is_available():
-            for cuda_device in numba.cuda.list_devices():
-                try:
-                    cuda_visible_gpus.append(
-                        self.get_device_by_cuda_index(cuda_device.id))
-                except TritonModelAnalyzerException:
-                    # Device not supported by DCGM, log warning
-                    logger.warning(
-                        f"Device '{str(cuda_device.name, encoding='ascii')}' with "
-                        f"cuda device id {cuda_device.id} is not supported by DCGM."
-                    )
-        return cuda_visible_gpus
+        nvmlInit()
+
+        gpu_devices = []
+        cuda_id_map = []
+        cuda_visiable_devices_set = "CUDA_VISIBLE_DEVICES" in os.environ
+        try:
+            if cuda_visiable_devices_set:
+                cuda_id_map = list(
+                    map(int,
+                        os.environ.get("CUDA_VISIBLE_DEVICES", "").split(",")))
```

Comment:
CUDA_VISIBLE_DEVICES also supports MIG IDs: https://docs.nvidia.com/datacenter/tesla/mig-user-guide/index.html#cuda-gi which are strings and not just ints.

You may be able to just keep it as a list of strings and instead check something like this:
```
if id in cuda_visible_devices or uuid in cuda_visible_devices:
   ...
```

**disclaimer**: I have no idea what values nvml will return for MIG devices, so you may want to just make this a FIXME or something to support MIG devices as needed based on priority.